sunpy/visualization/animator/wcs.py

```diff
@@ -0,0 +1,241 @@
+from functools import partial
+
+import numpy as np
+
+import astropy.units as u
+from astropy.wcs.wcsapi import BaseLowLevelWCS
+
+from sunpy.visualization.animator.base import ArrayAnimator
+
+
+__all__ = ['ArrayAnimatorWCS']
+
+
+class ArrayAnimatorWCS(ArrayAnimator):
+    """
+    Animate an array with associated `~astropy.wcs.wcsapi.BaseLowLevelWCS` object.
+
+    The following keyboard shortcuts are defined in the viewer:
+
+    * 'left': previous step on active slider.
+    * 'right': next step on active slider.
+    * 'top': change the active slider up one.
+    * 'bottom': change the active slider down one.
+    * 'p': play/pause active slider.
+
+    Parameters
+    ----------
+    data: `numpy.ndarray`
+        The data to be visualized.
+    wcs: `astropy.wcs.wcsapi.BaseLowLevelWCS`
+        The world coordinate object associated with the array.
+    slices: `tuple` or `list`
+        A list specifying which axes of the array should be plotted on which
+        axes. The list should be the same length as the number of pixel
+        dimensions with ``'x'`` and (optionally) ``'y'`` in the elements
+        corresponding to the axes to be plotted. If only ``'x'`` is present a
+        line plot will be drawn. All other elements should be `0`.
+    coord_params: `dict`, optional
+        This dict allows you to override
+        `~astropy.visualization.wcsaxes.WCSAxes` parameters for each world
+        coordinate. The keys of this dictionary should be a value which can be
+        looked up in ``WCSAxes.coords`` (i.e. ``em.wl`` or ``hpln``) and the
+        values should be a dict which supports the following keys, and passes
+        their values to the associated `~astropy.visualization.wcsaxes.WCSAxes`
+        methods.
+
+        * ``format_unit``: `~astropy.visualization.wcsaxes.CoordinateHelper.set_format_unit`
+        * ``major_formatter``: `~astropy.visualization.wcsaxes.CoordinateHelper.set_major_formatter`
+        * ``axislabel``: `~astropy.visualization.wcsaxes.CoordinateHelper.set_axislabel`
+        * ``grid``: `~astropy.visualization.wcsaxes.CoordinateHelper.grid` (The value should be a dict of keyword arguments to ``grid()`` or `True`).
+    ylim: `tuple` or `None`, optional
+       The yaxis limits to use when drawing a line plot, if not specified
+       defaults to the data limits.
+    ylabel: `string`, optional
+       The yaxis label to use when drawing a line plot. Setting the label on
+       the y-axis on an image plot should be done via ``coord_params``.
+
+    """
+
+    def __init__(self, data, wcs, slices, coord_params=None, ylim=None, ylabel=None, **kwargs):
+        if not isinstance(wcs, BaseLowLevelWCS):
+            raise ValueError("A WCS object should be provided that implements the astropy WCS API.")
+        if wcs.pixel_n_dim is not data.ndim:
```

Comment:
Should this not be ```wcs.pixel_n_dim =! data.ndim``` to keep it general to any number of dimensions?  (Although I realize if the dimensions were greater than the Python-pre-assigned ints the data cube will likely by too big for most systems...)

Reply:
yes, yes it should.